vic/drivers/cesm/cpl_esmf/lnd_comp_esmf.F90

```diff
@@ -0,0 +1,178 @@
+module lnd_comp_esmf
+
+! !USES:
+
+  use ESMF
+  use esmfshr_mod
+!
+! !PUBLIC TYPES:
+  implicit none
+  save
+  private ! except
+
+!--------------------------------------------------------------------------
+! Public interfaces
+!--------------------------------------------------------------------------
+
+  public :: lnd_init_esmf
+  public :: lnd_run_esmf
+  public :: lnd_final_esmf
+  public :: lnd_register_esmf
+
+!--------------------------------------------------------------------------
+! Private data interfaces
+!--------------------------------------------------------------------------
+
+!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+CONTAINS
+!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+subroutine lnd_register_esmf(comp, rc)
+    type(ESMF_GridComp)  :: comp
+    integer, intent(out) :: rc
+
+    rc = ESMF_SUCCESS
+
+    print *, "In lnd register routine"
```

Comment:
Per personal communication with @apcraig, I'm removing the esmf sources for now.  Since we didn't develop them for this work, we can always bring in the boilerplate again if we ever need them.